lte/gateway/c/oai/tasks/sgw/pgw_handlers.c

```diff
@@ -536,82 +549,137 @@ uint32_t pgw_handle_nw_initiated_bearer_deactv_req(
   while ((num_elements < hashtblP->num_elements) && (i < hashtblP->size) &&
-         (!found)) {
+         (!is_imsi_found)) {
     pthread_mutex_lock(&hashtblP->lock_nodes[i]);
     if (hashtblP->nodes[i] != NULL) {
       node = hashtblP->nodes[i];
       spgw_ctxt_p = node->data;
       num_elements++;
       if (spgw_ctxt_p != NULL) {
         if (!strcmp(
-              (const char *)
-                spgw_ctxt_p->sgw_eps_bearer_context_information.imsi.digit,
-              (const char *) bearer_req_p->imsi)) {
-          itti_s5_deactv_ded_bearer_req->s11_mme_teid =
+          (const char *)
+          spgw_ctxt_p->sgw_eps_bearer_context_information.imsi.digit,
+          (const char *) bearer_req_p->imsi)) {
+          is_imsi_found = true;
+          s11_mme_teid =
             spgw_ctxt_p->sgw_eps_bearer_context_information.mme_teid_S11;
-          for (j = 0; j < bearer_req_p->no_of_bearers; j++) {
-            if (
-              bearer_req_p->ebi[j] ==
-              spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection
-                .default_bearer) {
-              itti_s5_deactv_ded_bearer_req->delete_default_bearer = true;
-              found = true;
-              break;
+          if ((bearer_req_p->lbi != 0) &&
+            (bearer_req_p->lbi ==
+            spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection
+              .default_bearer)) {
+            is_lbi_found = true;
+            // Check if the received EBI is valid
+            for (itrn = 0; itrn < bearer_req_p->no_of_bearers; itrn ++) {
+              if(sgw_cm_get_eps_bearer_entry(
+                &spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection,
+                bearer_req_p->ebi[itrn])) {
+                is_ebi_found = true;
+                ebi_to_be_deactivated[no_of_bearers_to_be_deact] =
+                  bearer_req_p->ebi[itrn];
+                no_of_bearers_to_be_deact++;
+              } else {
+                invalid_bearer_id[no_of_bearers_rej] = bearer_req_p->ebi[itrn];
+                no_of_bearers_rej++;
+              }
             }
+          } else {
+            invalid_bearer_id[no_of_bearers_rej] = bearer_req_p->lbi;
+            no_of_bearers_rej++;
+            OAILOG_ERROR(
+              LOG_PGW_APP,
+              "Unknown LBI (%d) received in pgw_nw_init_deactv_bearer_request"
+              "\n",
+              bearer_req_p->lbi);
+            pthread_mutex_unlock(&hashtblP->lock_nodes[i]);
+            break;
           }
         }
       }
     }
     pthread_mutex_unlock(&hashtblP->lock_nodes[i]);
     i++;
   }
-  OAILOG_INFO(
-    LOG_PGW_APP,
-    "Sending nw_initiated_deactv_bearer_req to SGW"
-    "with delete_default_bearer flag set to %d\n",
-    itti_s5_deactv_ded_bearer_req->delete_default_bearer);
-  rc = itti_send_msg_to_task(TASK_SPGW_APP, INSTANCE_DEFAULT, message_p);
+
+  if ((!is_ebi_found) || (!is_lbi_found) || (!is_imsi_found)) {
```

Comment:
no_of_bearers_rej can be greater than 0, and all the flags are true. see comment above.
What shall we  do in such a case? send a reject and deactivate?
Please update the code and add comments to explain behaviour.

Reply:
In case of multiple bearers, if some EBIs are valid and some are not then we have to send reject to those which are invalid and proceed with deactivation for those which are valid. I have added this comment and added one more check for no_of_bearers_rej > 0 along with the flags.